src/client/QXmppClient.cpp

```diff
@@ -320,6 +324,27 @@ bool QXmppClient::isConnected() const
     return d->stream->isConnected();
 }
 
+/// Returns true if the current client state is "active", false if it is "inactive"
+/// See xep-0352.
+
+bool QXmppClient::clientState()
```

Comment:
I am not sure about the member name, or using a boolean to convey this information. Code like this does not seem very readable:

if (client->clientState()) {
..
}